src/sdk/strategy/pre-strategy.ts

```diff
@@ -180,16 +180,20 @@ export class PreStrategy {
 }
 
 export class DeployedPreStrategy {
-  constructor(
-    public label: string,
-    public cohort: Cohort,
-    public policy: EnactedPolicy,
-    public encrypter: Enrico,
-    public decrypter: PreTDecDecrypter,
-    private bobSecretKey: SecretKey,
-    public conditionSet?: ConditionSet
+  public constructor(
+    public readonly label: string,
+    public readonly cohort: Cohort,
+    public readonly policy: EnactedPolicy,
+    public readonly encrypter: Enrico,
+    public readonly decrypter: PreTDecDecrypter,
```

Comment:
Do these need to be passed in the constructor, or can they be created based on other values already passed to the constructor?

It also seems that the encrypter is made on the fly anyway via `makeEncrypter(...)`.

Reply:
Same question for DeployedCbdStrategy.